File: Pandacap.UI.Elements/IPost.vb
Imports Pandacap.UI.Badges

''' <summary>
''' A post that can be shown in one of Pandacap's "paged" areas, like the gallery or inbox.
''' </summary>
Public Interface IPost
    ''' <summary>
    ''' A badge that shows the origin of a remote post.
    ''' </summary>
    ReadOnly Property Badge As Badge

    ''' <summary>
    ''' The title to be shown for this post in a paged view.
    ''' </summary>
    ReadOnly Property DisplayTitle As String

    ''' <summary>
    ''' An opaque ID for this post; used in pagination.
    ''' </summary>
    ReadOnly Property Id As String

    ''' <summary>
    ''' A URL where the Pandacap administrator can view this content.
    ''' </summary>
    ReadOnly Property InternalUrl As String

    ''' <summary>
    ''' A URL where unauthenticated users can view this content.
    ''' </summary>
    ReadOnly Property ExternalUrl As String

    ''' <summary>
    ''' The date/time at which this content was posted or added.
    ''' </summary>
    ReadOnly Property PostedAt As DateTimeOffset

    ''' <summary>
    ''' The URL of the profile page of the user who posted this content, if any.
    ''' </summary>
    ReadOnly Property ProfileUrl As String

    ''' <summary>
    ''' A list of thumbnails associated with this content. Can be an empty list.
    ''' </summary>
    ReadOnly Property Thumbnails As IEnumerable(Of IPostThumbnail)

    ''' <summary>
    ''' The name of the user who posted this content.
    ''' </summary>
    ReadOnly Property Username As String

    ''' <summary>
    ''' An avatar associated with the user who posted this content, if any.
    ''' </summary>
    ReadOnly Property Usericon As String
End Interface

File: Pandacap.UI.Elements/IPost.cs
using Pandacap.UI.Badges;

namespace Pandacap.UI.Elements
{
    /// <summary>
    /// A post that can be shown in one of Pandacap's "paged" areas, like the gallery or inbox.
    /// </summary>
    public interface IPost
    {
        /// <summary>
        /// A badge that shows the origin of a remote post.
        /// </summary>
        Badge Badge { get; }

        /// <summary>
        /// The title to be shown for this post in a paged view.
        /// </summary>
        string DisplayTitle { get; }

        /// <summary>
        /// An opaque ID for this post; used in pagination.
        /// </summary>
        string Id { get; }

        /// <summary>
        /// A URL where the Pandacap administrator can view this content.
        /// </summary>
        string? InternalUrl { get; }

        /// <summary>
        /// A URL where unauthenticated users can view this content.
        /// </summary>
        string? ExternalUrl { get; }

        /// <summary>
        /// The date/time at which this content was posted or added.
        /// </summary>
        DateTimeOffset PostedAt { get; }

        /// <summary>
        /// The URL of the profile page of the user who posted this content, if any.
        /// </summary>
        string? ProfileUrl { get; }

        /// <summary>
        /// A list of thumbnails associated with this content. Can be an empty list.
        /// </summary>
        IEnumerable<IPostThumbnail> Thumbnails { get; }

        /// <summary>
        /// The name of the user who posted this content.
        /// </summary>
        string? Username { get; }

        /// <summary>
        /// An avatar associated with the user who posted this content, if any.
        /// </summary>
        string? Usericon { get; }
    }
}
